ADD: possibility to use CMAKE_C_FLAGS=" "

elsagermann opened the pull request:
## Status
**IN DEVELOPMENT**

## Description
The compilation of the omnicompiler uses a c99 compiler. The possibility to use the CMAKE_C_FLAGS in order to set the "-stc=c99" would be therefore useful in case one does not use a c99 compiler by default. The variable is then passed to the omnicompiler Autotools file via the CFLAGS=${CMAKE_C_FLAGS} command added to the main CMake file.

## Related issues
Issue #00 
Issue #01 

[100%] Performing build step for 'omni-compiler'
Created c-exprcode.h, c-exprcode.c
updating c-parser.h
updating c-parser.output
Created c-token.c, c-token.h
c-expr.c: In function ‘innerFreeExprOfBinaryNode’:
c-expr.c:2168:5: error: ‘for’ loop initial declarations are only allowed in C99 mode
     for(int i = 0; i < 2; ++i)
     ^
c-expr.c:2168:5: note: use option -std=c99 or -std=gnu99 to compile your code

clementval commented:
@elsagermann Which compiler at CSCS/MCH triggers this? The Jenkins plans was testing all 4 compilers and none triggered this back then? Just curious.

clementval commented on CMakeLists.txt:
```diff
@@ -91,6 +91,7 @@ if(BUILD_OMNI)
       && CC=${CMAKE_C_COMPILER} CXX=${CMAKE_CXX_COMPILER}
       FC=${CMAKE_Fortran_COMPILER} JAVAC=${Java_JAVAC_EXECUTABLE}
       ${OMNI_COMPILER_SUBMODULE}/configure
+      CFLAGS=${CMAKE_C_FLAGS}
```
Can you move it before `configure` just to be constituent with other definition like `JAVAC`/`FC` ...

elsagermann commented:
@clementval  Which compiler at CSCS/MCH triggers this? The Jenkins plans was testing all 4 compilers and none triggered this back then? Just curious.
The default gnu compiler on arolla, when using PrgEnv-cray/1.0.4. I changed the plan in order to use the buildenv script and added claw to it that's why it was green until then :)

clementval commented:
@elsagermann Ok so they still have GNU < 5 :-) ... Should think to update at some point
